nixos/modules/programs/rivalcfg.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.rivalcfg;
+in {
+  options.programs.rivalcfg = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Small CLI utility program that allows you to configure SteelSeries Rival gaming mice. Add yourself to the "mouse" group to be able to use rivalcfg as a user.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment = {
+      systemPackages = with pkgs; [ rivalcfg ];
+    };
+
+    users.extraGroups.mouse.gid = config.ids.gids.mouse;
+    services.udev.extraRules = fileContents "${pkgs.rivalcfg}/share/rivalcfg/99-steelseries-rival.rules";
```

Comment:
This is no good, this requires IFD to evaluate, you should use `udev.packages` or so instead